beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -2855,6 +2855,15 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         ChainSegmentResult::Successful { imported_blocks }
     }
 
+    /// Updates fork-choice node into a permanent `available` state so it can become a viable head.
+    /// Only completed sampling results are received. Blocks are unavailable by default and should
+    /// be pruned on finalization, on a timeout or by a max count.
+    pub async fn process_sampling_completed(self: &Arc<Self>, block_root: Hash256) {
+        // TODO(das): update fork-choice
```

Comment:
I think it would be possible that sampling is completed before block is imported into fork choice - we'll need to remember handling this scenario.

Reply:
I'll add a comment